api/envoy/api/v2/core/health_check_logging.proto

```diff
@@ -0,0 +1,71 @@
+syntax = "proto3";
+
+package envoy.api.v2.core;
+
+import "envoy/api/v2/core/base.proto";
+
+import "google/protobuf/duration.proto";
+import "google/protobuf/wrappers.proto";
+
+import "validate/validate.proto";
+import "gogoproto/gogo.proto";
+
+option (gogoproto.equal_all) = true;
+
+// [#protodoc-title: Health check logging]
+// :ref:`Health check logging <arch_overview_health_check_logging>`.
+// [#proto-status: draft]
+
+message HealthCheckEvent {
+  HealthCheckerType health_checker_type = 1;
+  string host_address = 2 [(validate.rules).string.min_bytes = 1];
+  string cluster_name = 3 [(validate.rules).string.min_bytes = 1];
+
+  oneof event {
+    option (validate.required) = true;
+
+    // Host ejection.
+    HealthCheckEjectUnhealthy eject_unhealthy_event = 4;
+
+    // Host addition.
+    HealthCheckAddHealthy add_healthy_event = 5;
+  }
+}
+
+enum HealthCheckFailureType {
+  ACTIVE = 0;
+  PASSIVE = 1;
+  NETWORK = 2;
+}
+
+enum HealthCheckerType {
+  HTTP = 0;
+  TCP = 1;
+  GRPC = 2;
+  REDIS = 3;
+}
+
+message HealthCheckEjectUnhealthy {
+  // The type of failure that caused this ejection
+  HealthCheckFailureType failure_type = 1;
+
+  // The timeout after which health checks fail for hosts in this cluster
+  google.protobuf.Duration timeout = 2 [(validate.rules).duration.required = true];
```

Comment:
It's a little strange to include timeout when we also log Active/Passive failures. Additionally, we aren't logging all the other types of timeouts that have been added. Unless we are going to log times specific to this event, I would probably drop timeout/unhealthy_threshold from this log since it can be intuited from the config.